### PR TITLE
BUG: Behaviour change in 1.5.0 when using Timedelta as Enum data type

### LINKED ISSUE #48908: Behaviour change in 1.5.0 when using `Timedelta` as `Enum` data type
From 1.5.0 it appears that `pd.Timedelta` behavior has changed when used as an `Enum` data-type...

#### Before:
```
>>> import pandas as pd   
>>> pd.__version__        
'1.4.3'
>>> from enum import Enum
>>> class TDEnum(pd.Timedelta, Enum):
...     T1 = 60000000000
...                             
...     def plus_one_hour(self): 
...         return self + pd.Timedelta(hours=1)
... 
>>> TDEnum["T1"]  # as is usual when defining a data-type, returns enum...
<TDEnum.T1: Timedelta('0 days 00:01:00')>
>>> # ...such that methods defined on the enum are available...
>>> TDEnum["T1"].plus_one_hour()
Timedelta('0 days 01:01:00')
```
#### 1.5.0:
```
>>> import pandas as pd   
>>> pd.__version__        
'1.5.0'
>>> # class defined as before...
>>> from enum import Enum
>>> class TDEnum(pd.Timedelta, Enum):
...     T1 = 60000000000
...                             
...     def plus_one_hour(self): 
...         return self + pd.Timedelta(hours=1)
... 
>>> TDEnum["T1"]  # now returns the pd.Timedelta enum value
Timedelta('0 days 00:01:00')
>>> # i.e. methods defined on the enum are no longer available
>>> TDEnum["T1"].plus_one_hour()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Timedelta' object has no attribute 'plus_one_hour'
```
I was reluctant to report this as a bug, although it is true that the `Timedelta` class appears to no longer behave as would be expected in this case (the usual behavior when defining a data-type (including `datetime.timedelta`) is as the 1.4.3 behavior described above).

If this is considered a bug, any chance it can be fixed?

If it's not a bug, might you have any ideas as to how I could replicate the previous behavior?

Thank you!

### PR DESCRIPTION
**Pull request by krasch**
- [X] closes #48908
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
pls add tests where appropriate

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -1502,7 +1502,7 @@ cdef class _Timedelta(timedelta):
     @classmethod
     def _from_value_and_reso(cls, int64_t value, NPY_DATETIMEUNIT reso):
         # exposing as classmethod for testing
-        return _timedelta_from_value_and_reso(value, reso)
+        return _timedelta_from_value_and_reso(cls, value, reso)
```
Comment (krasch) [inline pandas/_libs/tslibs/timedeltas.pyx:1505]: @jbrockmendel Is it still useful to have the `_timedelta_from_value_and_reso` function? After my changes it now has exactly the same signature as `_from_value_and_reso`

**Comment by krasch** [comment]
> pls add tests where appropriate

done

File: pandas/_libs/tslibs/timedeltas.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timedeltas.pyx:1505]: i _think_ its more performant as it is bc cython doesn't support cdef classmethods. havent done any measurements on this though

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 2]
```diff
@@ -1829,6 +1829,7 @@ class Timedelta(_Timedelta):
                 return NaT
 
             return _timedelta_from_value_and_reso(
+                Timedelta,
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/timedeltas.pyx:1832]: should this be `type(self)`?

Comment (krasch) [reply pandas/_libs/tslibs/timedeltas.pyx:1832]: I had added a question re: this but somehow github ate it, weird..

Anyway, here is my concern, if I make the change you are suggesting

```
td1 = MyTimedeltaSubclass("10 minutes")
td2 = Timedelta ("10 minutes")

then result types will be inconsistent
td1 + td2 -> MyTimedeltaSubclass
td2 + td1 -> Timedelta
```

Maybe that is fine, but it also kinda weirded me out, which is why I wanted to ask you. It feels a little bit like it should be the responsibility of the subclass to implement the custom __add__ behaviour. 

Wondering: does the rest of pandas respect subclasses in this way?

Comment (krasch) [reply pandas/_libs/tslibs/timedeltas.pyx:1832]: Should I go ahead and create an issue for this?

**Review by mroeschke — APPROVED** [review]
Looks reasonable. @jbrockmendel merge when ready

**Review by jbrockmendel — APPROVED** [review]

**Comment by jbrockmendel** [comment]
thanks @krasch

**Pull request merged**